### PR TITLE
Refresh blockchain checkpoints

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

**Comment by rnicoll** [comment]
Completed full reindexes on both mainnet and testnet with this patch, both to verify the checkpoints and to verify the overall build.

**Comment by rnicoll** [comment]
Ignore last, re-running mainnet sync now to re-validate

**Comment by rnicoll** [comment]
Okay, re-reindexed, all good

File: src/chainparams.cpp [hunk 1]
```diff
@@ -198,15 +198,16 @@ class CMainParams : public CChainParams {
...
-            // Data as of block 954c7c66dee51f0a3fb1edb26200b735f5275fe54d9505c76ebd2bcabac36f1e (height 3606083).
+            // Data as of block e4b4ecda4c022406c502a247c0525480268ce7abbbef632796e8ca1646425e75 (height 3854173).
             // Tx estimate based on average of year 2021 (~40k transactions per day)
-            1613217680, // * UNIX timestamp of last checkpoint block
-            71035468,   // * total number of transactions between genesis and last checkpoint
+            1628934997, // * UNIX timestamp of last checkpoint block
+            77616340,   // * total number of transactions between genesis and last checkpoint
                         //   (the tx=... number in the SetBestChain debug.log lines)
-            0.46        // * estimated number of transactions per second after checkpoint
+            0.31        // * estimated number of transactions per second after checkpoint
```
Comment (patricklodder) [inline src/chainparams.cpp:210]: `(77616340 - 71035468) / (1628934997 - 1613217680) = 0.4187019960213311`, so I think this should be `0.42`?

**Review by patricklodder — CHANGES_REQUESTED** [review]
I suspect that there's a typo in mainnet tx estimate.

File: src/chainparams.cpp (hunk 1 above)
Comment (rnicoll) [reply src/chainparams.cpp:210]: I've done it off a block about a week ago, tx rate has been dropping.

Comment (patricklodder) [reply src/chainparams.cpp:210]: which block? the checkpoint is 660 blocks old.

Comment (rnicoll) [reply src/chainparams.cpp:210]: Went back ~10,080 blocks, didn't think to keep the exact details sadly though.

Comment (rnicoll) [reply src/chainparams.cpp:210]: Trying block #3844093, 2edc6830e6e45acd45bf1d2354ead85fd22233258c19aa48d48db6f671346bd3, tx 77432592 at 1628299921

(77616340 - 77432592) / (1628934997 - 1628299921) = 183748 635076 = 0.2893

That seem a reasonable estimate?

Comment (rnicoll) [reply src/chainparams.cpp:210]:
```suggestion
            0.29        // * estimated number of transactions per second after checkpoint
```

Comment (patricklodder) [reply src/chainparams.cpp]: Not really. Previously we have done projections based on an entire year, now you're proposing to do it based on just a week of data. I think the sample set is too small, plus we're incentivizing an increase of on-chain transactions with this release, so I'd expect an increase in tx volume.

It's not that important though, I thought you were making a typo but if you think this is it and the last week is a good sample of what's to come in the future... then who am I to argue? Time will tell.

**Comment by rnicoll** [comment]
Happy to do a year, will do momentarily.

**Comment by rnicoll** [comment]
At block `6f23e77027a4c95f69fcd48f01ff5c8e2a9b36f2d1b50b30f00fa16333b2e188` from Aug 15, 2020 at 11:03 there were 63,193,574 transactions. Plugging that into the formula I get:

`(77616340 - 63193574) / (1628934997 - 1597489398) = 14422766 / 31445599 = 0.46`

I'll revert to the original figure, seems like it's stable over an extended period and I should ignore minor variation.

**Review by patricklodder — APPROVED** [review]
ACK.

- Checked checkpoints to be in active chain tip on all my testnet and mainnet nodes, all agree ✅ 
- numTx estimate is now done consistently w/ the past ✅
- checked for forks within fork window (1440 blocks since 3854990, which was the tip at the time of writing) for mainnet as the checkpoint could technically still be reversed on other 1.14 clients - only 2 forks of length 1 have been found across all my mainnet nodes ✅
    - ba5de51f9039cf6cfa36e302c686431440852abc02a9ce8c046af2a046e42dab at 3854469
    - b9de77ba10c8e329169e564bbea6fa194e5bf4965f4d676c4848930c8483de31 at 3854834
    - both seem to be a miner losing a single race for a block, so I expect the checkpoint to not be challenged.

**Pull request merged**